addons/controllers/calico/calicoconfig_controller.go

```diff
@@ -27,15 +45,39 @@ type CalicoConfigReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CalicoConfig object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
 func (r *CalicoConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("calicoconfig", req.NamespacedName)
+	ctxCancel, cancel := context.WithTimeout(ctx, contextTimeout)
+	defer cancel()
+	log := r.Log.WithValues("CalicoConfig", req.NamespacedName)
+
+	log.Info("Start reconciliation")
+
+	// fetch CalicoConfig resource, ignore not-found errors
+	calicoConfig := &cniv1alpha1.CalicoConfig{}
+	if err := r.Client.Get(ctxCancel, req.NamespacedName, calicoConfig); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// config resources are expected to have the same name as the cluster. However, we ideally try to read the cluster name from the owner reference of the addon config object
+	clusterName := req.NamespacedName.Name
+	for _, ownerRef := range calicoConfig.GetOwnerReferences() {
+		if ownerRef.APIVersion == clusterapiv1beta1.GroupVersion.String() && ownerRef.Kind == "Cluster" {
+			clusterName = ownerRef.Name
+			break
+		}
+	}
+
+	// verify that the cluster with the same name is present
+	cluster := &clusterapiv1beta1.Cluster{}
+	if err := r.Client.Get(ctxCancel, types.NamespacedName{Namespace: req.Namespace, Name: clusterName}, cluster); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
```

Comment:
question: do we want to do some logging here?

Reply:
Agreed, Done.